packages/manager/src/views/Settings/components/Symlink/CreateSymlink/CreateSymlink.tsx

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { Controller, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  Split,
+  SplitItem,
+  TextInput
+} from '@patternfly/react-core';
+import { useGetWebPropertyGroupedByEnv } from '@app/services/persistent';
+
+export const schema = yup.object({
+  // TODO: change this to URL validation, after server supports http protocol append
+
+  source: yup
+    .string()
+    .label('Source Directory')
+    .trim()
+    .required()
+    .test('source', 'Source directory entered is incorrect', (value) =>
+      /^[_.a-zA-Z0-9/-]+$/.test(value?.toString() || '')
```

Comment:
`/^[_.a-zA-Z0-9/-]+$/` store it in a constant

Reply:
done